java/java.lsp.server/nbcode/integration/src/org/netbeans/modules/nbcode/integration/layer.xml

```diff
@@ -39,7 +39,7 @@
             </folder>
             <file name="Import_hidden"/>
             <folder name="Maven2">
-                <file name="JavaApplication_hidden"/>
+                <file name="maven.archetype_hidden"/>
```

Comment:
Why was existing `JavaApplication` file renamed? Or where the `maven.archetype` file/folder comes from.

Reply:
Hm. I thought that `maven.archetype` was a way how to allow LSP to create a project. This is now replaced by the `JavaApplication2` template.

Re. rename: that's a question how to handle such situation: the original template is a **file**. I need to provide a folder structure, so:
* the template can be no longer a **file**, but must turn to a folder. Existing installations or extensions that may have contributed to that file in the ConfigFS would break (impossible to layer-merge file + folder).
* a new template will be introduced, which I did (JavaApplication2).
* a "contents here" attribute will be introduced. So the project itself can continue as **file** template, but its contents will be elsewhere. I even made code support for it, but at the end decided not to use it in the template.

Please share your thoughts.

Reply:
Isn't the `maven.archetype` registered only in `java/java.lsp.server/nbcode` suite? If so, it doesn't have to be hidden.

Changing `file` to `folder` has some drawback, your solution is fine. Consider calling the folder `JavaApp` to avoid the `2`.